client/my-sites/media/controller.js

```diff
@@ -31,4 +32,26 @@ export default {
 		} );
 		next();
 	},
+
+	singleMedia: function( context, next ) {
+		if ( ! getSiteFragment( context.path ) ) {
+			return page.redirect( '/media' );
+		}
+		// Render
+		context.primary = (
+			<SingleMediaComponent mediaId={ parseInt( context.params.mediaId ) } view="single" />
+		);
+		next();
+	},
+
+	editSingleMedia: function( context, next ) {
+		if ( ! getSiteFragment( context.path ) ) {
```

Comment:
If we're performing this check multiple times, perhaps it makes sense to move it to a separate controller middleware/callback and reuse it for each route as needed.